src/__test__/App.test.js

```diff
@@ -0,0 +1,51 @@
+import { render, screen } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import store from '../redux/store';
+
+import App from '../App';
+import Homepage from '../components/pages/Homepage';
+import Details from '../components/pages/Details';
+
+jest.mock('../components/pages/Homepage');
+jest.mock('../components/pages/Details');
+
+test('should render the Homepage', () => {
+  Homepage.mockImplementation(() => <h2>Welcome From Home Page</h2>);
+  render(
+    <MemoryRouter initialEntries={['/']}>
+      <Provider store={store}>
+        <App />
+      </Provider>
+    </MemoryRouter>,
+  );
+  const txt = screen.getByText('Welcome From Home Page');
+  expect(txt).toBeInTheDocument();
+});
+
+test('should render the Details Page', () => {
+  Details.mockImplementation(() => <h2>Angola From Details Page</h2>);
+  render(
+    <MemoryRouter initialEntries={['/details/Angola']}>
+      <Provider store={store}>
+        <App />
+      </Provider>
+    </MemoryRouter>,
+  );
+  const txt = screen.getByText('Angola From Details Page');
+  expect(txt).toBeInTheDocument();
+});
+
+describe('Test app.js with pages', () => {
+  test('should render the Header and Layout components', () => {
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
+    const txt = screen.getByText('Weather App');
+    expect(txt).toBeInTheDocument();
+  });
+});
```

Comment:
- [ ] Wonderful job for this project. You really put in nice effort. However, there're issues with the test for mocking both the Homepage and Details pages. Kindly give a closer look and fix where the issues are.

![test-fail-1](https://user-images.githubusercontent.com/26597432/218771690-4aee38dd-88c4-49e8-9c45-48630434f602.png)

![test-fail-2](https://user-images.githubusercontent.com/26597432/218771733-112eeab4-4a67-4d06-ad52-a0a42e06a788.png)